workers/unity/Packages/com.improbable.gdk.core/Worker/ConnectionHandlers/ConnectionFlows.cs

```diff
@@ -0,0 +1,372 @@
+                throw new AuthenticationFailedException("Did not receive a player identity token.");
+            }
+
+            if (result.Value.Status.Code != ConnectionStatusCode.Success)
+            {
+                throw new AuthenticationFailedException("Failed to retrieve a player identity token.\n" +
+                    $"error code: {result.Value.Status.Code}\nerror message: {result.Value.Status.Detail}");
+            }
+
+            return result.Value.PlayerIdentityToken;
+        }
+
+        /// <summary>
+        ///     Retrieves the login tokens for all active deployments that the player
+        ///     can connect to via the development authentication flow.
+        /// </summary>
+        /// <param name="workerType">The type of the worker that wants to connect.</param>
+        /// <param name="playerIdentityToken">The player identity token of the player that wants to connect.</param>
+        /// <returns>A list of all available login tokens and their deployments.</returns>
+        protected virtual List<LoginTokenDetails> GetDevelopmentLoginTokens(string workerType, string playerIdentityToken)
+        {
+            var result = DevelopmentAuthentication.CreateDevelopmentLoginTokensAsync(
+                RuntimeConfigDefaults.LocatorHost,
+                RuntimeConfigDefaults.AnonymousAuthenticationPort,
+                new LoginTokensRequest
+                {
+                    WorkerType = workerType,
+                    PlayerIdentityToken = playerIdentityToken,
+                    UseInsecureConnection = false,
+                    DurationSeconds = 120,
+                }
+            ).Get();
+
+            if (!result.HasValue)
+            {
+                throw new AuthenticationFailedException("Did not receive any login tokens back.");
+            }
+
+            if (result.Value.Status.Code != ConnectionStatusCode.Success)
+            {
+                throw new AuthenticationFailedException("Failed to retrieve any login tokens.\n" +
+                    $"error code: {result.Value.Status.Code}\nerror message: {result.Value.Status.Detail}");
+            }
+
+            return result.Value.LoginTokens;
+        }
+
+        /// <summary>
+        ///     Selects which login token to use to connect via the development authentication flow.
+        /// </summary>
+        /// <param name="loginTokens">A list of available login tokens.</param>
+        /// <returns>The selected login token.</returns>
+        protected virtual string SelectLoginToken(List<LoginTokenDetails> loginTokens)
+        {
+            if (loginTokens.Count == 0)
+            {
+                throw new AuthenticationFailedException("Did not receive any login tokens. Do you have a valid deployment running?");
+            }
+
+            return loginTokens[0].LoginToken;
+        }
+
+        /// <summary>
+        ///     Gets the player id for the player trying to connect via the development authentication flow.
```

Comment:
```suggestion
        ///     Gets the player ID for the player trying to connect via the development authentication flow.
```

id --> ID if you find any more